lib/sms.js

```diff
@@ -0,0 +1,86 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+var Nexmo = require('nexmo')
+var P = require('bluebird')
+
+var TEMPLATE_NAMES = new Map([
+  [ 1, 'installFirefox' ]
+])
+
+module.exports = function (log, translator, templates, smsConfig) {
+  var nexmo = new Nexmo({
+    apiKey: smsConfig.apiKey,
+    apiSecret: smsConfig.apiSecret
+  })
+  var sendSms = P.promisify(nexmo.message.sendSms, { context: nexmo.message })
+  var firefoxAppStoreLink = smsConfig.firefoxAppStoreLink
+
+  return {
+    send: function (phoneNumber, senderId, messageId, acceptLanguage) {
+      log.info({
+        op: 'sms.send',
+        senderId: senderId,
+        messageId: messageId,
+        acceptLanguage: acceptLanguage
+      })
+
+      return P.resolve()
+        .then(function () {
+          var message = getMessage(messageId, acceptLanguage)
+
+          return sendSms(senderId, phoneNumber, message.trim())
+        })
+        .then(function (result) {
+          var resultCount = result.messages && result.messages.length
+          if (resultCount !== 1) {
+            log.error({ op: 'sms.send', err: new Error('Unexpected result count'), resultCount: resultCount })
+          }
+
+          result = result.messages[0]
+          var status = result.status
+
+          if (status !== '0') {
+            var reason = result['error-text']
+            fail('Message rejected', {
+              status: 500,
+              reason: reason,
+              reasonCode: status
+            })
+          }
```

Comment:
Should we log a success here? Looks like auth-server logs a flowEvent, but maybe still useful to visualize success in Kibana.